docs/source/cluster_management/kvs-migration.rst

```diff
@@ -0,0 +1,91 @@
+.. _kvs-migration:
+
+=============
+KVS Migration
+=============
+
+.. danger::
+
+    Ensure you have backups before attempting a KVS migration.
+    Double check that `migrateConfig` (the target KVS) is specified correctly, since all tables in that KVS will be dropped.
+
+Overview
+========
+
+Migration of all transactional data from one KVS to another can be performed using the :ref:`AtlasDB CLI<clis>`.
+The source KVS (which contains the data to migrate) should be specified in the ``fromConfig``.
+The target KVS (which we will migrate to) should be specified in the ``migrateConfig``.
+
+There following are AtlasDB tables with special behaviour with respect to KVS migration:
+
+- Atomic tables, as specified in ``AtlasDbConstants.ATOMIC_TABLES``, are internal tables that use check and set and are not transactional.
+
+- Hidden tables, as specified in ``AtlasDbConstants.HIDDEN_TABLES``, are internal tables that are not transactional.
+
+- Cassandra hidden tables, as specified in ``HiddenTables.HIDDEN_TABLES``, are internal tables used by Cassandra KVS and are not transactional.
+
+- Targeted sweep tables, as specified in ``TargetedSweepSchema``, none of which are transactional.
+
+- Migration checkpoint table, which is a special table used for checkpointing during the migration.
+
+None of the above tables can be migrated, but due to legacy reasons the exact behaviour is more involved.
+The migration is performed in three steps which must be run in order: setup, followed by migrate, and finally validate.
+
+The first step in running any of the three commands is to take out a fresh timestamp in the source KVS, fast-forward the target KVS to a larger timestamp, and then take out two fresh timestamps on the target KVS for the migration start timestamp and the migration commit timestamp.
+Then, we immediately insert an entry into the target KVS with the above two timestamps, effectively commiting all transactions with the migration start timestamp.
+
+.. warning::
+
+    KVS migration must be run while AtlasDB is offline.
+    If you are using TimeLock, the TimeLock server must be running in order to do the migration.
+    Otherwise, you must use the ``--offline`` flag, which will remove the leader block from your configuration.
```

Comment:
remove the leader block from your configuration for the purposes of the migration